src/key.js

```diff
@@ -491,11 +491,11 @@ function getExpirationTime(keyPacket, selfCertificate) {
 Key.prototype.getPrimaryUser = function() {
   var primUser = [];
   for (var i = 0; i < this.users.length; i++) {
-    if (!this.users[i].userId || !this.users[i].selfCertifications) {
+    if ((!this.users[i].userId && !this.users[i].userAttribute) || !this.users[i].selfCertifications) {
```

Comment:
This changes the semantic of the method. I think getPrimaryUser should only return users with userId. In the case that userId is initial and userAttribute is set the user should be filtered out.